pkg/virt-operator/install-strategy/create.go

```diff
@@ -1339,89 +1339,57 @@ func createOrUpdateService(kv *v1.KubeVirt,
-		if !exists {
-			continue
-		} else if err != nil {
-			return err
-		}
-
-		privSCC, ok := privSCCObj.(*secv1.SecurityContextConstraints)
-		if !ok {
-			return fmt.Errorf("couldn't cast object to SecurityContextConstraints: %+v", privSCCObj)
-		}
-
-		oldUsers := privSCC.Users
-		privSCCCopy := privSCC.DeepCopy()
-		users := privSCCCopy.Users
-
-		modified := false
+	clientset kubecli.KubevirtClient,
+	expectations *util.Expectations) error {
+	sec := clientset.SecClient()
 
-		// remove users from previous
-		if curSccPriv != nil && !addOnly {
-			for _, acc := range curSccPriv.ServiceAccounts {
-				shouldRemove := true
-				// only remove if the target doesn't contain the same
-				// rule, otherwise leave as is.
-				for _, targetAcc := range sccPriv.ServiceAccounts {
-					if acc == targetAcc {
-						shouldRemove = false
-						break
-					}
-				}
-				if shouldRemove {
-					removed := false
-					users, removed = remove(users, acc)
-					modified = modified || removed
-				}
-			}
-		}
+	version := kv.Status.TargetKubeVirtVersion
+	imageRegistry := kv.Status.TargetKubeVirtRegistry
+	id := kv.Status.TargetDeploymentID
+	kvkey, err := controller.KeyFunc(kv)
+	if err != nil {
+		return err
+	}
 
-		// add any users from target that don't already exist
-		for _, acc := range sccPriv.ServiceAccounts {
-			if !contains(users, acc) {
-				users = append(users, acc)
-				modified = true
-			}
+	for _, scc := range targetStrategy.sccs {
+		var cachedSCC *secv1.SecurityContextConstraints
+		scc := scc.DeepCopy()
+		obj, exists, _ := stores.SCCCache.GetByKey(scc.Name)
+		if exists {
+			cachedSCC = obj.(*secv1.SecurityContextConstraints)
 		}
 
-		if modified {
-			oldUserBytes, err := json.Marshal(oldUsers)
+		injectOperatorMetadata(kv, &scc.ObjectMeta, version, imageRegistry, id)
+		if !exists {
+			expectations.SCC.RaiseExpectations(kvkey, 1, 0)
+			_, err := sec.SecurityContextConstraints().Create(scc)
 			if err != nil {
-				return err
+				expectations.SCC.LowerExpectations(kvkey, 1, 0)
+				return fmt.Errorf("unable to create SCC %+v: %v", scc, err)
 			}
-			userBytes, err := json.Marshal(users)
+			log.Log.V(2).Infof("SCC %v created", scc.Name)
+		} else if !objectMatchesVersion(&cachedSCC.ObjectMeta, version, imageRegistry, id) {
+			var ops []string
+			sccBytes, err := json.Marshal(scc)
 			if err != nil {
 				return err
 			}
-
-			test := fmt.Sprintf(`{ "op": "test", "path": "/users", "value": %s }`, string(oldUserBytes))
-			patch := fmt.Sprintf(`{ "op": "replace", "path": "/users", "value": %s }`, string(userBytes))
-
-			_, err = scc.SecurityContextConstraints().Patch(sccPriv.TargetSCC, types.JSONPatchType, []byte(fmt.Sprintf("[ %s, %s ]", test, patch)))
+			ops = append(ops, fmt.Sprintf(`{ "op": "replace", "path": "/", "value": %s }`, string(sccBytes)))
+			// replace the SCC by the new one
+			_, err = sec.SecurityContextConstraints().Patch(scc.Name, types.JSONPatchType, generatePatchBytes(ops))
```

Comment:
i'm not 100% sure this Patch on '/' will work because there's going to be things in the cached SCC's metadata (like creationTImestamp and ResourceVersion) that might cause issues if we attempt to patch them. 

I think it's probably safe to just call an Update() instead of using Patch() for the scc. So this could be as simple as something like this. 

```
scc.ObjectMeta = cachedScc.ObjectMeta.DeepCopy()
injectOperatorMetadata(kv, &scc.ObjectMeta, version, imageRegistry, id)
_, err = sec.SecurityContextConstraints().Update(scc)
```

Reply:
Changed to update as suggested, the update is safe.